src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -19,6 +19,7 @@
 using Raven.Server.Documents.Includes;
 using Raven.Server.Documents.Indexes.Auto;
 using Raven.Server.Documents.Indexes.MapReduce.Auto;
+using Raven.Server.Documents.Indexes.MapReduce.Static;
```

Comment:
Can you explain the idea behind this commit? I'm not sure that I'm following.
If we know the fields ahead of time, we go ahead and define the `Dictionary<string, IndexField> MapFields`, but how does this help us?

What about indexes that uses `CreateFields` ?